### PR TITLE
Java: Promote MVEL injection query from experimental

### PR DESCRIPTION
**Pull request by atorralba**
PR to promote the MVEL Injection query created in #3329

## Changes

* Existing files were moved out of experimental.
* The `MvelInjectionLib.qll` file was renamed to `MvelInjection.qll` and refactored to use the CSV sink model.
* Refactored tests to use `InlineExpectationsTest`.

## Evaluation

With the addition of the [JsonParser taint step introduced in the promotion of the OGNL injection query](https://github.com/github/codeql/pull/5894/files/1815656a02470d84b3f0054a64ae0d9238600ada..43d4575359def5b33940f245a89cec0780278e22), this query now detects [CVE-2020-13942](https://unomi.apache.org/security/cve-2020-13942.txt) (Remote Code Execution in Apache Unomi). This vulnerability was fixed with a filter that implements a deny-list of methods that can be called from MVEL, so the query will still report the issue even though Unomi is no longer vulnerable. Nonetheless, I tried adding that filter as a sanitizer and the query then correctly reports no issues, but it's such a specific sanitizer that I considered it wasn't worth to add it.

### CODE REVIEW

**Review by aschackmull — COMMENTED** [review]
I'd like some reshuffling of the content of the two files `MvelInjectionQuery.qll` and `MvelInjectionSinkModels.qll`. Only the configuration should go in `MvelInjectionQuery.qll` whereas the rest of it should go in a file called `MvelInjection.qll`, which can then both expose the extension points and be generally importable without leaking the configuration.

The qhelp also needs doc review.

**Comment by atorralba** [comment]
> I'd like some reshuffling of the content of the two files `MvelInjectionQuery.qll` and `MvelInjectionSinkModels.qll`. Only the configuration should go in `MvelInjectionQuery.qll` whereas the rest of it should go in a file called `MvelInjection.qll`, which can then both expose the extension points and be generally importable without leaking the configuration.

I was following the pattern applied to JEXL Injection in all my open PRs:

* https://github.com/github/codeql/blob/6823855e9cac61a72b934a2b8b1f99b246d656f1/java/ql/src/semmle/code/java/security/JexlInjectionSinkModels.qll
* https://github.com/github/codeql/blob/6823855e9cac61a72b934a2b8b1f99b246d656f1/java/ql/src/semmle/code/java/security/JexlInjectionQuery.qll

Should that be changed everywhere to `*Query.qll` containing only the configuration, and `*.qll` (instead of `*SinkModels.qll`) containing everything else and being imported in `ExternalFlow.qll`?

There are also some queries that use a secondary data flow configuration (e.g. `DataFlow2`). With this changes, should those be made public and moved to `*Query.qll` too?

**Comment by aschackmull** [comment]
There is some potential merit in keeping the sink models separate such that `ExternalFlow` doesn't need to import a lot of unrelated things - if we do that then we likely need 3 supporting `qll` files rather than 2, since all the extension points for the query shouldn't be in `*Query.qll`, but rather in something more generally importable. But I think the structure I outlined and which is current in this PR is also fine. Splitting the sinks into a third file doesn't really make much difference (the only potential difference might be compilation time). So yeah, the JEXL example should be refactored as well - we don't want the query extension points in a `*Query.qll` file. @smowton what do you think? Do you have a preference for 2 or 3 supporting qll files for this case?

File: java/ql/src/semmle/code/java/security/MvelInjection.qll [hunk 1]
```diff
@@ -0,0 +1,233 @@
+/** Provides classes to reason about MVEL injection attacks. */
+
+import java
+import semmle.code.java.dataflow.DataFlow
+import semmle.code.java.dataflow.ExternalFlow
```
Comment (aschackmull) [inline java/ql/src/semmle/code/java/security/MvelInjection.qll:5]: Let's keep these imports private.
```suggestion
private import semmle.code.java.dataflow.DataFlow
private import semmle.code.java.dataflow.ExternalFlow
```

File: java/ql/src/semmle/code/java/security/MvelInjection.qll [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+  )
+}
+
+/**
+ * Methods in `MvelScriptEngine` that compile a MVEL expression.
+ */
+private class MvelScriptEngineCompilationMethod extends Method {
+  MvelScriptEngineCompilationMethod() {
+    getDeclaringType() instanceof MvelScriptEngine and
+    (hasName("compile") or hasName("compiledScript"))
```
Comment (aschackmull) [inline java/ql/src/semmle/code/java/security/MvelInjection.qll:187]: Minor stylistic simplification:
```suggestion
    hasName(["compile", "compiledScript"])
```

**Review by aschackmull — COMMENTED** [review]
LGTM, but needs docs review of the qhelp as this wasn't done when it was originally merged to experimental.

**Review by mchammer01 — APPROVED** [review]
@atorralba - this is looking good to me ✨ 
I made a few minor tweaks to the qhelp file and noticed `powerfull` on line 23. Not sure whether this is a typo or a genuine context name, so left it as is for now.
Hope this helps!

**Comment by atorralba** [comment]
> @atorralba - this is looking good to me ✨
> I made a few minor tweaks to the qhelp file and noticed `powerfull` on line 23. Not sure whether this is a typo or a genuine context name, so left it as is for now.
> Hope this helps!

Thank you for the review! It was a typo indeed. I removed the word altogether since it wasn't too relevant, and added a "GOOD" example to the sample Java code.

**Review by artem-smotrakov — APPROVED** [review]
I am glad to see the query is going to be added to the main query pack!

**Review by aschackmull — APPROVED** [review]

**Pull request merged**